include/oneapi/dpl/pstl/hetero/dpcpp/utils_ranges_sycl.h

```diff
@@ -127,6 +131,15 @@ struct all_view_fn
     {
         return ::std::forward<_R>(__r);
     }
+#if _ONEDPL_CPP20_RANGES_PRESENT
+    template <typename _T, sycl::usm::alloc _MemoryType>
+    auto
+    operator()(const std::vector<_T, sycl::usm_allocator<_T, _MemoryType>>& __vec) const 
```

Comment:
Do I get it right that this function will make underlying data read-only because of `const`? Should there be a second version without `const`?

Reply:
in given case `const` means that `operator()` doesn't modify members of `struct all_view_fn`, and it doesn't influence on arguments of `operator()`.
Indeed, see an example https://godbolt.org/z/T9M9xc6nh

Reply:
Sorry for the confusion. I meant `const` related to the argument (`const std::vector...`): https://godbolt.org/z/3c16d45q8

Reply:
I see that this function has been removed in [cdf839e](https://github.com/oneapi-src/oneDPL/pull/1457/commits/cdf839edb9c9cc4d51dc49299a43301c3a78a59c).